commit d1eabdfa05d023d99dc3400ac60adb915f252111 (refs/changes/21/5621/2)
Author: Josh Wilsdon <josh@wilsdon.ca>
Date:   2019-02-25T10:24:00-08:00 (8 months ago)
    
    TRITON-1234 cn-agent should have task that returns current sysinfo
    Reviewed by: Orlando Vazquez <orlando@joyent.com>
    Approved by: Orlando Vazquez <orlando@joyent.com>

diff --git a/lib/backends/dummy/index.js b/lib/backends/dummy/index.js
index 3294b8f..1db6362 100644
--- a/lib/backends/dummy/index.js
+++ b/lib/backends/dummy/index.js
@@ -76,7 +76,7 @@ var queueDefns = [
         name: 'server_tasks',
         tasks: [
             'command_execute',
-            'server_sysinfo_load',
+            'server_sysinfo',
             'server_update_nics'
         ]
     }
diff --git a/lib/backends/dummy/tasks/server_sysinfo.js b/lib/backends/dummy/tasks/server_sysinfo.js
new file mode 100644
index 0000000..1e6ab7b
--- /dev/null
+++ b/lib/backends/dummy/tasks/server_sysinfo.js
@@ -0,0 +1,45 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/*
+ * Copyright (c) 2019, Joyent, Inc.
+ */
+
+var assert = require('assert-plus');
+var common = require('../common');
+var SysinfoGetter = require('../lib/sysinfo');
+var Task = require('../../../task_agent/task');
+
+var ServerSysinfoTask = module.exports = function (req) {
+    Task.call(this);
+    this.req = req;
+};
+
+Task.createTask(ServerSysinfoTask);
+
+function start(callback) {
+    var self = this;
+
+    assert.object(self.req, 'self.req');
+    assert.object(self.req.sysinfo, 'self.req.sysinfo');
+    assert.object(self.req.serverAddress, 'self.req.serverAddress');
+
+    (new SysinfoGetter()).get({
+        serverAddress: self.req.serverAddress,
+        serverUuid: self.req.sysinfo.UUID
+    }, function _onSysinfo(err, sysinfoObj) {
+        if (err) {
+            self.fatal({error: err});
+            return;
+        }
+
+        self.finish({
+            sysinfo: sysinfoObj
+        });
+    });
+}
+
+ServerSysinfoTask.setStart(start);
diff --git a/lib/backends/smartos/index.js b/lib/backends/smartos/index.js
index 4d70ffd..2f6a97f 100644
--- a/lib/backends/smartos/index.js
+++ b/lib/backends/smartos/index.js
@@ -54,7 +54,8 @@ var queueDefns = [
         name: 'server_tasks',
         tasks: [
             'command_execute',
-            'server_overprovision_ratio'
+            'server_overprovision_ratio',
+            'server_sysinfo'
         ]
     },
     {
diff --git a/lib/backends/smartos/tasks/server_sysinfo.js b/lib/backends/smartos/tasks/server_sysinfo.js
new file mode 100644
index 0000000..c94da56
--- /dev/null
+++ b/lib/backends/smartos/tasks/server_sysinfo.js
@@ -0,0 +1,36 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/*
+ * Copyright (c) 2019, Joyent, Inc.
+ */
+
+var smartdc_config = require('../smartdc-config');
+var Task = require('../../../task_agent/task');
+
+var ServerSysinfoTask = module.exports = function (req) {
+    Task.call(this);
+    this.req = req;
+};
+
+Task.createTask(ServerSysinfoTask);
+
+function start(callback) {
+    var self = this;
+
+    smartdc_config.sysinfo(function _onSysinfo(err, sysinfoObj) {
+        if (err) {
+            self.fatal({error: err});
+            return;
+        }
+
+        self.finish({
+            sysinfo: sysinfoObj
+        });
+    });
+}
+
+ServerSysinfoTask.setStart(start);
diff --git a/package.json b/package.json
index 2ccc38d..26aa145 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "cn-agent",
   "description": "Triton Compute Node Agent",
-  "version": "2.9.0",
+  "version": "2.10.0",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
